specs/PushNotifications/PushNotifications-spec.md

```diff
@@ -0,0 +1,438 @@
+            // The CLSID associated with the Client COM server that reunion will adopt
+            String TaskClsid{ get; };
+
+            // The conditions under which Push Triggers would execute
+            Windows.ApplicationModel.Background.IBackgroundCondition Condition;
+
+            // The in-memory Event handler to subscribe to if the app is in the foreground
+            event Windows.Foundation.EventHandler<PushNotificationActivatedEventArgs> PushActivated;
+
+            // The backgroundRegistration that gets populated on successful registration
+            Windows.ApplicationModel.Background.IBackgroundTaskRegistration Registration;
+        };
+
+        enum ChannelStatus
+        {
+            InProgress, // The request is in progress and there is no retry operation
+            InProgressRetry, // The request is in progress and is in a backoff retry state
+            CompletedSuccess, // The request completed successfully
+            CompletedFailure, // The request failed with some critical internal error. Check ExtendedError for HRESULT
+        };
+
+        runtimeclass PushNotificationChannelResult
+        {
+            PushNotificationChannelResult(
+                Windows.Networking.PushNotifications.PushNotificationChannel channel,
+                HRESULT extendedError,
+                ChannelStatus status);
+
+            // The Push channel associated with the Result. Null if InProgress or completion failed
+            Windows.Networking.PushNotifications.PushNotificationChannel Channel { get; };
+
+            // Low-level information about why the operation might not
+            // have worked. Apps generally do not take action based on
+            // this value, in part because the value may not be stable.
+            HRESULT ExtendedError{ get; };
+
+            // The Status of the ChannelComplete operation
+            ChannelStatus Status { get; };
+        };
+
+        static runtimeclass PushNotificationChannelManager
+        {
+            // Register an activator using a Details context
+            static void RegisterPushNotificationActivator(InProcActivatorDetails details);
+
+            // Unregister any activator if present
+            static void UnregisterPushNotificationActivator(Windows.ApplicationModel.Background.IBackgroundTaskRegistration registration);
+
+            // Request a Push Channel with an encoded RemoteId from WNS. RemoteId is an AAD identifier GUID
+            static Windows.Foundation.IAsyncOperationWithProgress<PushNotificationChannelResult, PushNotificationChannelResult> CreatePushNotificationChannelAsync(String remoteId);
+        };
+    }
+```
+# Appendix
+* For all OS images before Cobalt, the RemoteId will be a noop and the native platform will send the PFN (PackageFamilyName) in the channel request protocol. The WNS server will maintain an internal mapping of PFN -> RemoteId and will return a channelUri with the encoded RemoteId after querying the map.
```

Comment:
`Cobalt` has no external meaning, can you please provide public build numbers here.

Reply:
@andrewleader Whats best practise for stating API and OS limitations? Should we simply go with build number?

Reply:
Yes, build number is the right answer

P.S. I often refer to https://en.wikipedia.org/wiki/Windows_10_version_history#Channels when I need to xref release dates, build numbers, codenames, etc. One of several such lists but one of the more comprehensive, nice layout and I can easily remember how to find it ;-)

Reply:
Thanks so this would essentially map to 21h1 or 19043 where we added the feature set. @sharath2727 Can you confirm that the private API you introduced for RemoteId can be supported with 21H1 or if there is more work to be done here?

Reply:
@hulumane Apologies for the late response here. It should be supported. We are not blocking this under any feature key here. We should be able to call into the private APIs without any extra effort. However, I have to follow up on your previous email (offline), on the instructions to call into the private APIs. Will update this space as soon as I figure out everything. Thanks.